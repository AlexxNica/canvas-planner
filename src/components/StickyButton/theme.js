import { darken } from 'instructure-ui/lib/util/color';

export default function generator ({ borders, colors, typography, spacing }) {
  return {
    fontSize: typography.fontSizeXSmall,
    fontFamily: typography.fontFamily,
    fontWeight: typography.fontWeightNormal,
    color: colors.white,
    background: colors.brand,
    backgroundHover: darken(colors.brand, 5),
    padding: `0 ${spacing.small}`,
    textTransform: 'uppercase',
    lineHeight: spacing.medium,
    iconMargin: spacing.xxSmall,
    hasIconRightPadding: spacing.xSmall,
    borderRadius: borders.radiusMedium,
    focusRingWidth: borders.widthSmall,
    focusRingColor: colors.brand
  };
}

generator.canvas = function (variables) {
  return {
    background: variables["ic-brand-primary"],
    backgroundHover: darken(variables["ic-brand-primary"], 5),
    focusRingColor: variables["ic-brand-primary"],
  };
};
