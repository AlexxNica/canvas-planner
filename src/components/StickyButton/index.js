import React, { Component } from 'react';
import classnames from 'classnames';
import themeable from 'instructure-ui/lib/themeable';
import { bool, func, node, number, oneOf } from 'prop-types';
import IconMoveUpLine from 'instructure-icons/lib/Line/IconMoveUpLine';
import IconMoveDownLine from 'instructure-icons/lib/Line/IconMoveDownLine';

import styles from './styles.css';
import theme from './theme.js';

class StickyButton extends Component {
  static propTypes = {
    children: node.isRequired,
    onClick: func,
    disabled: bool,
    direction: oneOf(['none', 'up', 'down']),
    zIndex: number
  };

  static defaultProps = {
    direction: 'none'
  };

  handleClick = (e) => {
    const {
      disabled,
      onClick
    } = this.props;

    if (disabled) {
      e.preventDefault();
      e.stopPropagation();
    } else if (typeof onClick === 'function') {
      onClick(e);
    }
  };

  renderIcon () {
    const direction = this.props.direction;

    if (direction === 'up') {
      return <IconMoveUpLine className={styles.icon} />;
    } else if (direction === 'down') {
      return <IconMoveDownLine className={styles.icon} />;
    } else {
      return null;
    }
  }

  render () {
    const {
      children,
      disabled,
      direction,
      zIndex
    } = this.props;

    const classes = {
      [styles.root]: true,
      [styles['direction--' + direction]]: direction !== 'none'
    };

    const style = {
      zIndex: (zIndex) ? zIndex : null
    };

    return (
      <button
        type="button"
        onClick={this.handleClick}
        className={classnames(classes)}
        style={style}
        aria-disabled={(disabled) ? 'true' : null}
      >
        <span className={styles.layout}>
          {children}
          {this.renderIcon()}
        </span>
      </button>
    );
  }
}

export default themeable(theme, styles)(StickyButton);
