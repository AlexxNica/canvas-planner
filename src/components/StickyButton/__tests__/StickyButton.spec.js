import React from 'react';
import StickyButton from '../index';
import {shallow} from 'enzyme';
import IconMoveUpLine from 'instructure-icons/lib/Line/IconMoveUpLine';
import IconMoveDownLine from 'instructure-icons/lib/Line/IconMoveDownLine';


it('renders', () => {
  const wrapper = shallow(<StickyButton>I am a Sticky Button</StickyButton>);
  expect(wrapper).toMatchSnapshot();
});

it('calls the onClick prop when clicked', () => {
  const fakeOnClick = jest.fn();
  const wrapper = shallow(
    <StickyButton onClick={fakeOnClick}>
      Click me
    </StickyButton>
  );

  wrapper.find('button').simulate('click');
  expect(fakeOnClick).toHaveBeenCalled();
});

it('does not call the onClick prop when disabled', () => {
  const fakeOnClick = jest.fn();
  const wrapper = shallow(
    <StickyButton onClick={fakeOnClick} disabled>
      Disabled button
    </StickyButton>
  );

  wrapper.find('button').simulate('click', {
    preventDefault() {},
    stopPropagation() {}
  });
  expect(fakeOnClick).not.toHaveBeenCalled();
});

it('renders the correct up icon', () => {
  const wrapper = shallow(
    <StickyButton direction="up">
      Click me
    </StickyButton>
  );
  expect(wrapper.find(IconMoveUpLine)).toHaveLength(1);
});

it('renders the correct down icon', () => {
  const wrapper = shallow(
    <StickyButton direction="down">
      Click me
    </StickyButton>
  );
  expect(wrapper.find(IconMoveDownLine)).toHaveLength(1);
});
