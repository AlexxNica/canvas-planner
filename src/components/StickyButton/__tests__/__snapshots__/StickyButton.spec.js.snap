// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<button
  aria-disabled={null}
  className=""
  onClick={[Function]}
  style={
    Object {
      "zIndex": null,
    }
  }
  type="button"
>
  <span>
    I am a Sticky Button
  </span>
</button>
`;
